### PR TITLE
Fix unhandled rejection if errors occur in a committed transaction. Fixes #3689

### LINKED ISSUE #3689: (PostgreSQL) Automanaged transaction with `{isolationLevel: 'SERIALIZABLE'}` throws an error.
The error is `Unhandled rejection Error: commit has been called on this transaction(<transactionId>), you can no longer use it.`

Without much investigation, I'd say it looks like that's because `commit` throws an error if I run two concurrent queries and they conflict. The first one commits OK and the other one throws error on `commit` - I assume the transaction manager then catches the error and tries to `rollback`.

### PR DESCRIPTION
**Pull request by BridgeAR**

### CODE REVIEW

**Comment by BridgeAR** [comment]
This fixes #3689

File: test/integration/transaction.test.js [hunk 1]
```diff
@@ -73,6 +73,30 @@ describe(Support.getTestDialectTeaser('Transaction'), function() {
         expect(t.finished).to.be.equal('rollback');
       });
     });
+
+    if (dialect === 'postgres' || dialect === 'mssql') {
```
Comment (mickhansen) [inline test/integration/transaction.test.js]: what's the issue with mysql here?

**Comment by mickhansen** [comment]
Looks good, just the single comment and a rebase.

File: test/integration/transaction.test.js (hunk 1 above)
Comment (BridgeAR) [inline test/integration/transaction.test.js]: Mysql returns the values and does not throw. But that does not really matter as my fix works for every dialect if something like that happens in that dialect.

**Pull request merged**